LICENSE

```diff
@@ -1,32 +1,202 @@
-Boost Software License - Version 1.0 - August 17th, 2003
-
-Permission is hereby granted, free of charge, to any person or organization
-obtaining a copy of the software and accompanying documentation covered by
-this license (the "Software") to use, reproduce, display, distribute,
-execute, and transmit the Software, and to prepare derivative works of the
-Software, and to permit third-parties to whom the Software is furnished to
-do so, all subject to the following:
-
-The copyright notices in the Software and this entire statement, including
-the above license grant, this restriction and the following disclaimer,
-must be included in all copies of the Software, in whole or in part, and
-all derivative works of the Software, unless such copies or derivative
-works are solely in the form of machine-executable object code generated by
-a source language processor.
-
-THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
-IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
-FITNESS FOR A PARTICULAR PURPOSE, TITLE AND NON-INFRINGEMENT. IN NO EVENT
-SHALL THE COPYRIGHT HOLDERS OR ANYONE DISTRIBUTING THE SOFTWARE BE LIABLE
-FOR ANY DAMAGES OR OTHER LIABILITY, WHETHER IN CONTRACT, TORT OR OTHERWISE,
-ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
-DEALINGS IN THE SOFTWARE.
-
----------------------------------------------------------------------------
-Note:
-Individual files contain the following tag instead of the full license text.
-
-	SPDX-License-Identifier:	BSL-1.0
-
-This enables machine processing of license information based on the SPDX
-License Identifiers that are here available: http://spdx.org/licenses/
```

Comment:
@Blast545 hmm I'm not entirely sure we can simply drop an existing license? @chapulina @tfoote for valuable feedback here.

Reply:
This makes sense to me, as Apache 2 is the license listed on `package.xml` for the contents of this package. It's a similar situation to https://github.com/ros2/tinydir_vendor/pull/12 and others.

But just to make sure: the contents of this package are indeed meant to be Apache, or are they Boost like the upstream code?

Reply:
Yes, we have used the package.xml to decide which is the valid license of the repo. Just to be sure maybe we can ask @nuclearsandwich (maintainer) or @wjwwood who uploaded the LICENSE file in the repo

Reply:
I think putting the Apache license here is ok, though it's not something we've really discussed in the past. The package.xml is clear about it:

https://github.com/ros2/poco_vendor/blob/546f40ac6ca3efcdc47ac5c244f37a3011b42c3f/package.xml#L10-L11

Reply:
This seems to be the pattern being followed by all vendor packages wrapping a package whose license is not Apache 2:

Reply:
Sure, my point is that I think you can safely remove the license here and replace it as it's obvious from the `package.xml` that we didn't intend to license the contents of the package under the boost license.